pkg/database/timescaledb/create.go

```diff
@@ -0,0 +1,231 @@
+package timescaledb
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/pkg/errors"
+	schemasv1alpha4 "github.com/schemahero/schemahero/pkg/apis/schemas/v1alpha4"
+	"github.com/schemahero/schemahero/pkg/database/postgres"
+)
+
+func CreateTableStatements(tableName string, tableSchema *schemasv1alpha4.TimescaleDBTableSchema) ([]string, error) {
+	postgresSchema := toPostgresTableSchema(tableSchema)
+	statements, err := postgres.CreateTableStatements(tableName, postgresSchema)
+	if err != nil {
+		return nil, errors.Wrap(err, "create postgres table statements")
+	}
+
+	// add the timecale extensions
+	if tableSchema.Hypertable != nil {
+		stmt, err := createHypertableStatement(tableName, tableSchema.Hypertable, tableSchema.Columns)
+		if err != nil {
+			return nil, errors.Wrap(err, "create hypertable statement")
+		}
+		if stmt != "" {
+			statements = append(statements, stmt)
+		}
+
+	}
+
+	// check for compression
+	if tableSchema.Hypertable.Compression != nil {
+		stmts, err := createCompressionStatements(tableName, tableSchema.Hypertable, tableSchema.Columns)
+		if err != nil {
+			return nil, errors.Wrap(err, "create compression statements")
+		}
+
+		statements = append(statements, stmts...)
+	}
+
+	// check for retention
+	if tableSchema.Hypertable.Retention != nil {
+		stmts, err := createRetentionStatements(tableName, tableSchema.Hypertable, tableSchema.Columns)
+		if err != nil {
+			return nil, errors.Wrap(err, "create retention statements")
+		}
+
+		statements = append(statements, stmts...)
+	}
+
+	return statements, nil
+}
+
+func createRetentionStatements(tableName string, hypertable *schemasv1alpha4.TimescaleDBHypertable, columns []*schemasv1alpha4.PostgresqlTableColumn) ([]string, error) {
+	if hypertable.Retention == nil {
+		return nil, nil
+	}
+
+	stmts := []string{}
+
+	if hypertable.Retention != nil {
+		if hypertable.Retention.Interval != "" {
+			if !isValidInterval(hypertable.Retention.Interval) {
+				return nil, errors.New("invalid interval")
+			}
+
+			stmt := fmt.Sprintf(`select add_retention_policy(%s, interval '%s'`, pgx.Identifier{tableName}.Sanitize(), hypertable.Retention.Interval)
```

Comment:
missing closing parenthesis?

Reply:
it's there, just doesn't yet know if there's more params.  this is a string builder thing going on